### PR TITLE
Add content negotiation

### PR DESCRIPTION
**Pull request by chiransiriwardhana**
## Purpose
Add accept and acceptLanguage fields to SubscriberServiceConfig annotation for content negotiation in the initial discovery request.

## Approach
Use annotation to retrieve accept and acceptLanguage fields.
Add retrieved values to initial discovery request.

## Samples
@websub:SubscriberServiceConfig {
path: "/ordereventsubscriber",
subscribeOnStartUp: true,
target: "http://localhost:9090/ordermgt/order",
accept:"application/json",
acceptLanguage:"de-DE",
leaseSeconds: 3600,
secret: "Kslk30SNF2AChs2"
}

### CODE REVIEW

File: websub-ballerina/commons.bal [hunk 1]
```diff
@@ -394,10 +396,13 @@ public function extractTopicAndHubUrls(http:Response response) returns @tainted
...
-
-    if (linkHeaders.length() == 0) {
+    if (response.statusCode == http:STATUS_NOT_ACCEPTABLE) {
+        return WebSubError("Content negotiation failed. Accept and/or Accept-Language headers mismatch");
+    } else if (response.statusCode == http:STATUS_INTERNAL_SERVER_ERROR) {
+        return WebSubError("Content negotiation failed. Accept and/or Accept-Language headers unavailable in discovery request");
+    } else if (linkHeaders.length() == 0) {
         return WebSubError("Link header unavailable in discovery response");
-    }
+    } else {}
```
Comment (Bhashinee) [inline websub-ballerina/commons.bal:405]: Shall we remove this else part?

File: websub-generic-integration-tests/tests/test_content_negotiation.bal [hunk 2]
```diff
@@ -0,0 +1,151 @@
+import ballerina/http;
```
Comment (Bhashinee) [inline websub-generic-integration-tests/tests/test_content_negotiation.bal:1]: Shall we add the license header?

File: websub-generic-integration-tests/tests/test_custom_subscribers.bal [hunk 3]
```diff
@@ -260,6 +260,7 @@ public class CustomSubWebhookServerForHeaderAndPayload {
 }
 
 @test:Config {}
+
```
Comment (Bhashinee) [inline websub-generic-integration-tests/tests/test_custom_subscribers.bal:263]: Shall we remove this line?

File: websub-ballerina/commons.bal (hunk 1 above)
Comment (chiransiriwardhana) [reply websub-ballerina/commons.bal]: okay

File: websub-generic-integration-tests/tests/test_custom_subscribers.bal (hunk 3 above)
Comment (chiransiriwardhana) [reply websub-generic-integration-tests/tests/test_custom_subscribers.bal:263]: okay

File: websub-generic-integration-tests/tests/test_content_negotiation.bal (hunk 2 above)
Comment (chiransiriwardhana) [reply websub-generic-integration-tests/tests/test_content_negotiation.bal:17]: okay

File: websub-generic-integration-tests/tests/test_content_negotiation.bal [hunk 4]
```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Bhashinee) [inline websub-generic-integration-tests/tests/test_content_negotiation.bal:1]: The year should be 2020.

**Pull request closed without merging**